subworkflows/local/pairalign_m2m/main.nf

```diff
@@ -37,17 +37,24 @@ workflow PAIRALIGN_M2M {
         ch_target
     )
 
-    // Train alignment parameters
+    // Train alignment parameters if not provided
     //
-    ALIGNMENT_TRAIN (
-        ch_queries,
-        ALIGNMENT_LASTDB.out.index.map { row -> row[1] }  // Remove metadata map
-    )
+    if (params.lastal_params) {
+        ch_queries_with_params = ch_queries.map { row -> [ row[0], row[1], file(params.lastal_params, checkIfExists: true) ] }
+        training_results_for_multiqc = channel.empty()
+    } else {
+        ALIGNMENT_TRAIN (
```

Comment:
Missing versions mixing - you must do this for every module/subworkflow itself, and not rely purely on the first module even if it's from the same collection of tools (theoretically you could have two versions of the same tool)

Reply:
Thanks for the heads up. I added the other modules versions channels and :boom: I found a bug in `last/train`, which fortunately was easy to fix.  I guess I need the following PR (https://github.com/nf-core/modules/pull/7393) to be merged before I can import the fix here.